### ISSUE TITLE
[core] Renaming template files causes errors

### ISSUE THREAD

**Issue by engelfrost** [pattern-lab/patternlab-node#887]
I am using Pattern Lab Node `v3.0.0-alpha.16` on `Mac`, with Node `v9.6.1`, using a `Vanilla` Edition.

##### Expected Behavior

* The pattern `avatar` is removed. 
* The pattern `profile-picture` is added.

##### Actual Behavior

* The pattern `avatar` is NOT removed.
* The pattern `profile-picture` is added.
* PL reports that:

```
TypeError: Cannot read property 'map' of undefined
    at PatternGraph.nodes2patterns (/Users/josef/Code/fabrikor/node_modules/@pattern-lab/core/src/lib/pattern_graph.js:238:18)
    at PatternGraph.lineage (/Users/josef/Code/fabrikor/node_modules/@pattern-lab/core/src/lib/pattern_graph.js:251:17)
    at cascadePatternStates (/Users/josef/Code/fabrikor/node_modules/@pattern-lab/core/src/lib/lineage_hunter.js:97:42)
    at Object.cascade_pattern_states (/Users/josef/Code/fabrikor/node_modules/@pattern-lab/core/src/lib/lineage_hunter.js:172:7)
    at Promise.all.then (/Users/josef/Code/fabrikor/node_modules/@pattern-lab/core/src/lib/buildPatterns.js:101:32)
    at <anonymous>
Error processing meta patterns
Error processing meta patterns
```

* On restart, `avatar` is removed but the "View all" view is broken (no patterns are shown). Individual patterns work. The message above is still produced. 
* Removing `dependencyGraph.json` resolves the issue

##### Steps to Reproduce

1. Rename a template (e.g. `atoms/media/avatar.mustache` => `atoms/media/profile-picture.mustache`.

Restarting PL generates a broken "View all" page.

**Comment by bmuenzenmeyer** [comment]
Additional debugging:

I get this output:

```
TypeError: Cannot set property 'compileState' of undefined
    at module.exports (D:\src\patternlab-node\packages\core\src\lib\compose.js:22:47)
    at patternsToBuild.map.pattern (D:\src\patternlab-node\packages\core\src\lib\buildPatterns.js:144:23)
    at Array.map (<anonymous>)
    at render.then.results (D:\src\patternlab-node\packages\core\src\lib\buildPatterns.js:143:64)
    at <anonymous>
Error rendering pattern lab header
Built Pattern Lab front end
```

**Comment by bmuenzenmeyer** [comment]
Workaround until fixed: re-generate using instructions above on file rename

**Comment by JosefBredereck** [comment]
The issue was released with v5.13.0 🎉
